etl/activities_to_socrata.py

```diff
@@ -0,0 +1,70 @@
+"""
+Downloads activities for CSRs data from a CSV report endpoint and then uploads the data to a Socrata dataset
+"""
+
+import os
+import logging
+
+import pandas as pd
+from sodapy import Socrata
+
+from field_maps import ACTIVITIES_MAP
+import utils
+
+# Socrata Secrets
+SO_WEB = os.getenv("SO_WEB")
+SO_TOKEN = os.getenv("SO_TOKEN")
+SO_KEY = os.getenv("SO_KEY")
+SO_SECRET = os.getenv("SO_SECRET")
+DATASET = os.getenv("ACTIVITIES_DATASET")
+
+# CSR CSV data endpoint
+ENDPOINT = os.getenv("ACTIVITIES_ENDPOINT")
+
+
+def transform(df):
+    logger.info("Transforming Activities")
+
+    # Field mapping
+    df = df[list(ACTIVITIES_MAP.keys())]
+    df.rename(columns=ACTIVITIES_MAP, inplace=True)
+
+    # date column formatting to match format expected by Socrata
+    df = utils.transform_datetime_formats(df)
+
+    # Remove rows without an activity id, this happens when a CSR has no activities associated with it
+    df = df.dropna(subset=["activity_id"])
```

Comment:
i am having trouble picturing this—you have rows of activities and some don't have IDs? as if they are placeholder/null activities?

Reply:
@johnclary, this is that the report looks like. Just a bunch of SR numbers and no activity ID/activity data. If you look these SRs up, you'll see they all have zero activities so I think this is a quirk of how 311 set up this report for us.
 
<img width="1166" alt="Screenshot 2024-08-13 at 12 10 13 PM" src="https://github.com/user-attachments/assets/addfe0e2-b241-4680-8bf1-27d53e29bf79">

Reply:
🙃 i see now thank youu